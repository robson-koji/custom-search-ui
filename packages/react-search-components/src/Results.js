import PropTypes from "prop-types";
import React from "react";

function Results({ children }) {
  return <ul>{children}</ul>;
}

Results.propTypes = {
  children: PropTypes.node.isRequired
};

export default Results;
