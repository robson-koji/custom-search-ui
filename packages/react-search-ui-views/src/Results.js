import PropTypes from "prop-types";
import React from "react";

import { appendClassName } from "./view-helpers";

function Results({ children, className }) {
  return (
    <ul className={appendClassName("sui-results-container", className)}>
      {children}
    </ul>
  );
}

Results.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Results;
