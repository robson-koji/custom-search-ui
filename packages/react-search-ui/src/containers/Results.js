import PropTypes from "prop-types";
import React, { Component } from "react";
import { Result, Results } from "@elastic/react-search-components";

import { withSearch } from "..";
import { Result as ResultType } from "../types";

function htmlEscape(str) {
  return String(str)
    .replace(/&/g, "&amp;")
    .replace(/"/g, "&quot;")
    .replace(/'/g, "&#39;")
    .replace(/</g, "&lt;")
    .replace(/>/g, "&gt;");
}

/*
  Our `Result` component expects result fields to be formatted in an object
  like:
  {
    field1: "value1",
    field2: "value2"
  }
*/
function formatResultFields(result) {
  return Object.keys(result.data).reduce((acc, n) => {
    // Fallback to raw values here, because non-string fields
    // will not have a snippet fallback. Raw values MUST be html escaped.
    let value = result.getSnippet(n) || htmlEscape(result.getRaw(n));
    value = Array.isArray(value) ? value.join(", ") : value;
    acc[n] = value;
    return acc;
  }, {});
}
export class ResultsContainer extends Component {
  static propTypes = {
    results: PropTypes.arrayOf(ResultType).isRequired,
    titleField: PropTypes.string,
    trackClickThrough: PropTypes.func,
    urlField: PropTypes.string
  };

  handleClickLink = id => {
    const { trackClickThrough } = this.props;
    !!trackClickThrough && trackClickThrough(id);
  };

  render() {
    const { results, titleField, urlField } = this.props;

    return (
      <Results>
        {results.map(result => (
          <Result
            fields={formatResultFields(result)}
            key={`result-${result.getRaw("id")}`}
            onClickLink={() => this.handleClickLink(result.getRaw("id"))}
            title={
              result.getSnippet(titleField) ||
              htmlEscape(result.getRaw(titleField))
            }
            url={result.getRaw(urlField)}
          />
        ))}
      </Results>
    );
  }
}

export default withSearch(ResultsContainer);
