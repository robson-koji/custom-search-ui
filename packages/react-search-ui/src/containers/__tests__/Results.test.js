import React from "react";
import { shallow } from "enzyme";
import { ResultsContainer } from "../Results";

const params = {
  results: [
    {
      data: { foo: "foo" },
      getRaw: val => val,
      getSnippet: val => `<em>${val}</em>`
    },
    {
      data: { bar: "bar" },
      getRaw: val => val,
      getSnippet: val => `<em>${val}</em>`
    }
  ],
  trackClickThrough: jest.fn(),
  titleField: "title",
  urlField: "url"
};

beforeEach(() => {
  params.trackClickThrough = jest.fn();
});

it("renders correctly", () => {
  const wrapper = shallow(<ResultsContainer {...params} />);
  expect(wrapper).toMatchSnapshot();
});

it("will call back when a document link is clicked in the view", () => {
  const wrapper = shallow(<ResultsContainer {...params} />);

  wrapper
    .find("Result")
    .at(0)
    .prop("onClickLink")();

  const clickThrough = params.trackClickThrough.mock.calls[0][0];
  expect(clickThrough).toEqual("id");
});
