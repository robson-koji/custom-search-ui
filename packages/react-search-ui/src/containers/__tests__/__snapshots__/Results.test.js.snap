// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Results>
  <Result
    fields={
      Object {
        "foo": "<em>foo</em>",
      }
    }
    key="result-id"
    onClickLink={[Function]}
    title="<em>title</em>"
    url="url"
  />
  <Result
    fields={
      Object {
        "bar": "<em>bar</em>",
      }
    }
    key="result-id"
    onClickLink={[Function]}
    title="<em>title</em>"
    url="url"
  />
</Results>
`;

exports[`supports a render prop 1`] = `
<div>
  <Result
    fields={
      Object {
        "foo": "<em>foo</em>",
      }
    }
    key="result-id"
    onClickLink={[Function]}
    title="<em>title</em>"
    url="url"
  />
  <Result
    fields={
      Object {
        "bar": "<em>bar</em>",
      }
    }
    key="result-id"
    onClickLink={[Function]}
    title="<em>title</em>"
    url="url"
  />
</div>
`;

exports[`supports an individual result render prop 1`] = `
Array [
  <li>
    &lt;em&gt;title&lt;/em&gt;
  </li>,
  <li>
    &lt;em&gt;title&lt;/em&gt;
  </li>,
]
`;
